### ISSUE TITLE
Failure building CoreCLR from ARM machine

### ISSUE THREAD

**Issue by ericstj** [dotnet/runtime#45087]
It looks like we assume that ARM architectures are always cross-targeted:
https://github.com/dotnet/runtime/blob/bf302de7f184c6e9dc23382980860af2849219e3/src/coreclr/crossgen-corelib.proj#L24-L27

I believe this first clause needs to be
```xml
      <CrossDir Condition="'$(BuildArchitecture)' != '$(TargetArchitecture)' and ('$(TargetArchitecture)' == 'arm' or '$(TargetArchitecture)' == 'arm64')">x64</CrossDir>
```

I made that change and was able to proceed further.  If folks think this is correct I can submit a PR.

**Comment by jkotas** [comment]
cc @trylek

**Comment by ericstj** [comment]
Related: https://github.com/dotnet/runtime/issues/39394

**Comment by AntonLapounov** [comment]
Did you hit this issue on Windows ARM64?  I believe we have fixed it while working on #49864.  Please reopen if you still see the issue.
